clang/lib/Driver/Compilation.cpp

```diff
@@ -47,7 +47,7 @@ Compilation::Compilation(const Driver &D, const ToolChain &_DefaultToolChain,
 Compilation::~Compilation() {
   // Remove temporary files. This must be done before arguments are freed, as
   // the file names might be derived from the input arguments.
-  if (!TheDriver.isSaveTempsEnabled() && !ForceKeepTempFiles)
+  if ((!TheDriver.isSaveTempsEnabled() || !TheDriver.isDumpDeviceCodeEnabled()) && !ForceKeepTempFiles)
```

Comment:
I tried to mimic the logic from save-temps, since we want the .`spv` files to be saved and not deleted after the compilation.
But I don't see the .spv saved with this logic. 
Is this approach ok?

Reply:
This approach seems aggressive to me.  Dumping device code isn't equivalent to -save-temps.  We should keep the behavior separate, what do you think?

Reply:
>> Dumping device code isn't equivalent to -save-temps.
It is not. I was just trying to see if I can save the .spv files that I was able to redirect using the --out-dir option in llvm-foreach.
If dump device code option is passed, the .spv files are put in the given path. With this code, I was trying to save them, otherwise, they get deleted when compilation exits.

Reply:
Also, save-temps, saves all the intermediate files.
But with dump device code, I'm only trying to save the .spv files.

Reply:
Okay - I think your logic is not quite right, try this
```suggestion
  if (!(TheDriver.isSaveTempsEnabled() || TheDriver.isDumpDeviceCodeEnabled()) && !ForceKeepTempFiles)
```

Reply:
Thank you! The .spv files now get saved in the given directory.